src/vs/workbench/services/search/node/textSearch.ts

```diff
@@ -5,330 +5,178 @@
-	private extraFiles: string[];
-	private maxResults: number;
+	private config: IRawSearch;
 	private walker: FileWalker;
-	private contentPattern: RegExp;
-	private isCanceled: boolean;
-	private isDone: boolean;
-	private total: number;
-	private worked: number;
-	private progressed: number;
 	private walkerError: Error;
-	private walkerIsDone: boolean;
-	private fileEncoding: string;
-	private limitReached: boolean;
+
+	private isCanceled = false;
+	private isDone = false;
+	private totalBytes = 0;
+	private processedBytes = 0;
+	private progressed = 0;
+	private walkerIsDone = false;
+	private limitReached = false;
+	private numResults = 0;
+
+	private nextWorker = 0;
+	private workers: ISearchWorker[] = [];
+	private workerClients: Client[] = [];
 
 	constructor(config: IRawSearch, walker: FileWalker) {
-		this.rootFolders = config.rootFolders;
-		this.extraFiles = config.extraFiles;
+		this.config = config;
 		this.walker = walker;
-		this.contentPattern = strings.createRegExp(config.contentPattern.pattern, config.contentPattern.isRegExp, { matchCase: config.contentPattern.isCaseSensitive, wholeWord: config.contentPattern.isWordMatch, multiline: false, global: true });
-		this.isCanceled = false;
-		this.limitReached = false;
-		this.maxResults = config.maxResults;
-		this.worked = 0;
-		this.progressed = 0;
-		this.total = 0;
-		this.fileEncoding = encodingExists(config.fileEncoding) ? config.fileEncoding : UTF8;
 	}
 
-	public cancel(): void {
+	cancel(): void {
 		this.isCanceled = true;
 		this.walker.cancel();
-	}
 
-	public search(onResult: (match: ISerializedFileMatch) => void, onProgress: (progress: IProgress) => void, done: (error: Error, complete: ISerializedSearchComplete) => void): void {
-		let resultCounter = 0;
+		this.workers.forEach(w => w.cancel());
+	}
 
-		let progress = () => {
-			this.progressed++;
-			if (this.progressed % Engine.PROGRESS_FLUSH_CHUNK_SIZE === 0) {
-				onProgress({ total: this.total, worked: this.worked }); // buffer progress in chunks to reduce pressure
+	search(onResult: (match: ISerializedFileMatch) => void, onProgress: (progress: IProgress) => void, done: (error: Error, complete: ISerializedSearchComplete) => void): void {
+		this.startWorkers();
+		const progress = () => {
+			if (++this.progressed % Engine.PROGRESS_FLUSH_CHUNK_SIZE === 0) {
+				onProgress({ total: this.totalBytes, worked: this.processedBytes }); // buffer progress in chunks to reduce pressure
 			}
 		};
 
-		let unwind = (processed: number) => {
-			this.worked += processed;
+		const unwind = (processed: number) => {
+			this.processedBytes += processed;
 
 			// Emit progress() unless we got canceled or hit the limit
 			if (processed && !this.isDone && !this.isCanceled && !this.limitReached) {
 				progress();
 			}
 
 			// Emit done()
-			if (this.worked === this.total && this.walkerIsDone && !this.isDone) {
+			if (!this.isDone && this.processedBytes === this.totalBytes && this.walkerIsDone) {
```

Comment:
No, we only check the file size once. This is based on the size that the fileWalker provides. I could change it to >= though.